index.js

```diff
@@ -274,10 +275,29 @@ function startSyncWithRetry(watcherArray, maximumNumberOfAttempts, pauseSeconds)
 }
 
 app.listen(config.net.port, () => {
-  startSyncWithRetry(watchers, config.start.attempts, config.start.pauseSeconds);
-
-  debug('finder ' + config.version.major + '.' + config.version.minor + '.' +
-    config.version.bug + ' with api version ' + config.version.api +
-    ' waiting for requests on port ' + config.net.port);
+  //delete existing index and create new index with spatial mapping 
+  esclient.indices.exists({ index: config.elasticsearch.index })
+  .then(function (resp) {
+    if (resp) { //if config.elasticsearch.index already exists
```

Comment:
This should at least be optional, and have a debug message!

I know you cannot always update a mapping, i.e. there might be incompatible changes between mappings. But if there are no incompatible changes, then there is no need to drop the index and recreate it.

So the more complex alternative would be "only recreat if the mapping is different/incompatible", but that's too complex.

Let settle with the following: Add a flag that the index is deleted and created at each startup (default) or not. If not, then the user has to take care that the mapping is not incompatible. Add another config variable to put the mapping at startup (default) or not.